tss-esapi/src/structures/tagged/schemes.rs

```diff
@@ -61,3 +69,424 @@ impl TryFrom<TPMT_KEYEDHASH_SCHEME> for KeyedHashScheme {
         }
     }
 }
+
+/// Enum representing the rsa scheme
+///
+/// # Details
+/// This corresponds to TPMT_RSA_SCHEME.
+/// This uses a subset of the TPMU_ASYM_SCHEME
+/// that has the TPMI_ALG_RSA_SCHEME as selector.
+#[derive(Clone, Copy, Debug)]
+pub enum RsaScheme {
+    RsaSsa(HashScheme),
+    RsaEs,
+    RsaPss(HashScheme),
+    Oaep(HashScheme),
+    Null,
+}
+
+impl RsaScheme {
+    /// Creates a new RsaScheme
+    pub fn create(
+        rsa_scheme_algorithm: RsaSchemeAlgorithm,
+        hashing_algorithm: Option<HashingAlgorithm>,
+    ) -> Result<RsaScheme> {
+        match rsa_scheme_algorithm {
+            RsaSchemeAlgorithm::RsaSsa => Ok(RsaScheme::RsaSsa(HashScheme::new(
+                hashing_algorithm.ok_or_else(|| {
+                    error!(
+                        "Hashing algorithm is required when creating RSA scheme of type RSA SSA"
+                    );
+                    Error::local_error(WrapperErrorKind::ParamsMissing)
+                })?,
+            ))),
+            RsaSchemeAlgorithm::RsaEs => Ok(RsaScheme::RsaEs),
```

Comment:
This seems to apply to a lot of enum variants, don't know if we want to go down this rabbit hole of always checking the other params are correct... I think we should, but maybe we can set that as a future issue to deal with.

Reply:
Yeah, I think you are right. First I thought that it might be enough to just give a warning. But then I reconsidered and liked the idea better to give an error.